packages/itwinui-react/src/core/Stepper/StepperStep.tsx

```diff
@@ -132,7 +137,7 @@ export const StepperStep = React.forwardRef((props, forwardedRef) => {
           {...circleProps}
           className={cx('iui-stepper-circle', circleProps?.className)}
         >
-          {index + 1}
+          {stepContent && isPast ? stepContent(index) : index + 1}
```

Comment:
We do not need to pass the `index` back into it `stepContent`, because it's not useful (the whole point is that the user wants to render something else).

Also I agree with @r100-stack's other comment.

```suggestion
          {stepContent ? stepContent() : index + 1}
```

Reply:
It should still be a function, so that the steps array can be defined outside a component. (There shouldn't be any lose React Nodes created outside a component; using a function allows us to delay the creation of these nodes)